queries/perl/highlights.scm

```diff
@@ -1,198 +1,107 @@
 
-; Keywords for including
 [ "use" "no" "require" ] @include
 
-; Keywords that mark conditional statements
 [ "if" "elsif" "unless" "else" ] @conditional
-(ternary_expression
-  ["?" ":"] @conditional.ternary)
 
-; Keywords that mark repeating loops
-[ "while" "until" "for" "foreach" ] @repeat
+(conditional_expression [ "?" ":" ] @conditional.ternary) 
 
-; Keyword for return expressions
-[ "return" ] @keyword.return
+[ "while" "until" "for" "foreach" ] @repeat
 
-; Keywords for phaser blocks
-; TODO: Ideally these would be @keyword.phaser but vim-treesitter doesn't
-;   have such a thing yet
-[ "BEGIN" "CHECK" "UNITCHECK" "INIT" "END" ] @keyword.function
+"return" @keyword.return
 
-; Keywords to define a function
-[ "sub" ] @keyword.function
+"sub" @keyword.function
 
-; Keywords that are regular infix operators
-[
-  "and" "or" "not" "xor"
-  "eq" "ne" "lt" "le" "ge" "gt" "cmp"
-] @keyword.operator
+[ "map" "grep" ] @function.builtin
 
-; Variables
 [
-  (scalar_variable)
-  (array_variable)
-  (hash_variable)
-] @variable
+  "package"
+  "do"
+  "my" "our" "local"
+  "last" "next" "redo" "goto"
+  "undef"
+] @keyword
 
-; Special builtin variables
-[
-  (special_scalar_variable)
-  (special_array_variable)
-  (special_hash_variable)
-  (special_literal)
-  (super)
-] @variable.builtin
+(_ operator: _ @operator)
+(yadayada) @exception
 
-((scalar_variable) @variable.builtin
- (#eq? @variable.builtin "$#ARGV"))
+(phaser_statement phase: _ @keyword.phaser)
 
-; Integer numbers
-[
-  (integer)
-  (hexadecimal)
-] @number
+[ "=>" "," ";" "->" ] @punctuation.delimiter
 
-; Float numbers
 [
-  (floating_point)
-  (scientific_notation)
-] @float
-
-; version sortof counts as a kind of multipart integer
-(version) @constant
-
-; Package names are types
-(package_name) @type
-
-; The special SUPER:: could be called a namespace. It isn't really but it
-;   should highlight differently and we might as well do it this way
-(super) @namespace
-
-; Comments are comments
-(comments) @comment @spell
-
-((source_file . (comments) @preproc)
-  (#lua-match? @preproc "^#!/"))
+  "or" "and"
+  "eq" "ne" "cmp" "lt" "le" "ge" "gt"
+  "isa"
+] @keyword.operator
 
-; POD should be handled specially with its own embedded subtype but for now
-;   we'll just have to do this.
-(pod_statement) @text
+(eof_marker) @preproc
+(data_section) @comment
 
-(method_invocation
-  function_name: (identifier) @method.call)
-(call_expression
-  function_name: (identifier) @function.call)
+(pod) @text
 
-; Built-in functions
-((call_expression
-  function_name: (identifier) @function.builtin)
- (#any-of? @function.builtin
-   "print" "printf" "sprintf" "say"
-   "push" "pop" "shift" "unshift" "splice"
-   "exists" "delete" "keys" "values"
-   "each"))
+(number) @number
+(version) @number
 
-[
-  (map)
-  (grep)
-  (bless)
-] @function.builtin
+(string_literal) @string
+(interpolated_string_literal) @string
+(quoted_word_list) @string
+(command_string) @string
```

Comment:
```suggestion
[
  (string_literal)
  (interpolated_string_literal)
  (quoted_word_list)
  (command_string)
] @string
```

Reply:
`(heredoc_content)` can also be grouped here.